tests/functional/base_test.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright 2023.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package functional
+
+import (
+	"fmt"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	manilav1 "github.com/openstack-k8s-operators/manila-operator/api/v1beta1"
+)
+
+func CreateManilaSecret(namespace string, name string) *corev1.Secret {
+	return th.CreateSecret(
+		types.NamespacedName{Namespace: namespace, Name: name},
+		map[string][]byte{
+			"ManilaPassword":         []byte(manilaTest.ManilaPassword),
+			"ManilaDatabasePassword": []byte(manilaTest.ManilaPassword),
+			"MetadataSecret":         []byte(manilaTest.ManilaPassword),
+		},
+	)
+}
+
+func CreateManilaMessageBusSecret(namespace string, name string) *corev1.Secret {
+	s := th.CreateSecret(
+		types.NamespacedName{Namespace: namespace, Name: name},
+		map[string][]byte{
+			"transport_url": []byte(fmt.Sprintf("rabbit://%s/fake", name)),
+		},
+	)
+	logger.Info("Secret created", "name", name)
+	return s
+}
+
+func CreateUnstructured(rawObj map[string]interface{}) *unstructured.Unstructured {
+	logger.Info("Creating", "raw", rawObj)
+	unstructuredObj := &unstructured.Unstructured{Object: rawObj}
+	_, err := controllerutil.CreateOrPatch(
+		ctx, k8sClient, unstructuredObj, func() error { return nil })
+	Expect(err).ShouldNot(HaveOccurred())
+	return unstructuredObj
+}
+
+func GetDefaultManilaSpec() map[string]interface{} {
+	return map[string]interface{}{
+		"databaseInstance": "openstack",
+		"secret":           SecretName,
+		"manilaAPI":        GetDefaultManilaAPISpec(),
+		"manilaScheduler":  GetDefaultManilaSchedulerSpec(),
+		"manilaShare":      GetDefaultManilaShareSpec(),
+	}
+}
+
+func GetDefaultManilaAPISpec() map[string]interface{} {
+	return map[string]interface{}{
+		"secret":         SecretName,
+		"replicas":       1,
+		"containerImage": manilav1.ManilaAPIContainerImage,
```

Comment:
If you enable the webhook in suite_test.go then you can drop this and let the webhook default it. Can be done in a follow up.

Reply:
Done here: https://github.com/openstack-k8s-operators/manila-operator/pull/77